spark3-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestExpireSnapshotsProcedure.java

```diff
@@ -163,18 +163,28 @@ public void testInvalidExpireSnapshotsCases() {
 
     AssertHelpers.assertThrows("Should reject calls without all required args",
         AnalysisException.class, "Missing required parameters",
-        () -> sql("CALL %s.system.expire_snapshots('n')", catalogName));
+        () -> sql("CALL %s.system.expire_snapshots()", catalogName));
 
     AssertHelpers.assertThrows("Should reject calls with invalid arg types",
-        RuntimeException.class, "Couldn't parse identifier",
+        AnalysisException.class, "Wrong arg type",
         () -> sql("CALL %s.system.expire_snapshots('n', 2.2)", catalogName));
 
-    AssertHelpers.assertThrows("Should reject empty namespace",
-        IllegalArgumentException.class, "Namespace cannot be empty",
-        () -> sql("CALL %s.system.expire_snapshots('', 't')", catalogName));
+    AssertHelpers.assertThrows("Should reject calls without empty table identifier",
+        IllegalArgumentException.class, "Cannot handle an empty identifier",
+        () -> sql("CALL %s.system.expire_snapshots('')", catalogName));
+  }
 
-    AssertHelpers.assertThrows("Should reject empty table name",
-        IllegalArgumentException.class, "Table name cannot be empty",
-        () -> sql("CALL %s.system.expire_snapshots('n', '')", catalogName));
+  @Test
+  public void testResolvingTableInAnotherCatalog() throws IOException {
+    String anotherCatalog = "another_" + catalogName;
+    spark.conf().set("spark.sql.catalog." + anotherCatalog, SparkCatalog.class.getName());
+    spark.conf().set("spark.sql.catalog." + anotherCatalog + ".type", "hadoop");
+    spark.conf().set("spark.sql.catalog." + anotherCatalog + ".warehouse", "file:" + temp.newFolder().toString());
+    spark.sql(String.format("CREATE TABLE %s.%s (id bigint NOT NULL, data string) USING iceberg", anotherCatalog,
```

Comment:
nit: we can probably use our `sql()` helper method to get parameterization for free.